### ISSUE TITLE
Errors when using flatten_annotations

### ISSUE THREAD

**Issue by earthlingworks** [gettalong/hexapdf#179]
Hello! We're seeing some errors using `flatten_annotations` on some PDFs.

One of the errors:

```
NoMethodError (undefined method `to_ary' for #<HexaPDF::Type::XRefStream:0x00007fbec27acdc0>)
```

The other error: 

```
NoMethodError (undefined method `configuration' for <HexaPDF::Document:77520>:HexaPDF::Document)
```

These errors were there in the previous version and still there in the current version. I'll send the PDFs through email. Thanks!

**Comment by gettalong** [comment]
Thanks for the files - I will have a look!

**Comment by gettalong** [comment]
The error `NoMethodError (undefined method `configuration' for <HexaPDF::Document:77520>:HexaPDF::Document)` is a bug in HexaPDF which I will fix. However, I couldn't reproduce it using your provided file.

**Comment by earthlingworks** [comment]
@gettalong ok, thanks. Sorry, I just realized that I forgot to provide the script to reproduce. Here it is in case it's helpful:

```
    document = HexaPDF::Document.open('/path/file.pdf')
    document.pages.each(&:flatten_annotations)
    document.write('/path/file2.pdf', validate: false, optimize: true)
```

**Comment by gettalong** [comment]
@earthlingworks Thanks - can confirm that the second error is fixed. Will do a bug fix release.

**Comment by earthlingworks** [comment]
Great; thank you!

**Comment by gettalong** [comment]
@earthlingworks I just released version 0.22.0 with the fix.

**Comment by earthlingworks** [comment]
Thank you!
